### DISCUSSION TITLE
Can I use Github code spaces and still be able to run the RNW App?

### DISCUSSION THREAD

**Discussion by AlmTalal** [microsoft/react-native-windows#12185] [Q&A] [answered]
Hello ! I just saw an [article](https://eveningkid.medium.com/github-codespaces-for-react-native-61d450db8fc0) which explains how to run a react native expo app in Github code spaces.

The advantage that code spaces can give to my company is that we will have to set up the environment just once, and we will not have the headaches that we are currently having every time a new member joins the team. So I thought that Github Code Spaces could be of help. 

The thing is that I don't know if it's possible to run and debug the application with GCS since GSC are virtual machines running projects hosted on GitHub (so maybe I will be able to see the web debugger but not the complete app). 

Has anyone done this before, what has been your experience ?

**Comment by chrisglein** [comment, marked as answer]
The magic that makes that work for mobile is that your GitHub Code Space is running in your browser (talking to some VM) and then via Expo you can run your app on a device (after scanning the QR code). However Expo doesn't work on Windows, so that key space is missing. For Windows the app would need to run... somewhere. And I don't think Github Code Spaces has anything other than your VSCode window, right?

You might be able to run Metro in your Code Space and then locally on a Windows machine configure the app to connect to that remote Metro instance (after figuring out your connection settings). That'd get you that fast refresh hosted dev environment, but you'd still need a Windows machine somewhere (could be a separate VM?) to connect to it, because I don't think you can create that within the Code Space.

TLDR, haven't tried that before but no I don't think it'd work as cleanly due to lack of Expo support for Windows.
